charts/internal/azure-infra/templates/main.tf

```diff
@@ -73,17 +73,75 @@ resource "azurerm_network_security_group" "workers" {
   {{- end}}
 }
 
+{{ if .Values.create.natGateway -}}
+#===============================================
+#= NAT Gateway
+#===============================================
+{{ if .Values.natGateway -}}
```

Comment:
See other comment - probably it makes sense to remove this code from this PR as it is unused now.